include/alpaka/mem/buf/BufCpu.hpp

```diff
@@ -519,6 +518,11 @@ namespace alpaka
                         if(!mem::buf::isPinned(buf))
                         {
 #if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && defined(__CUDACC__)
+                            if(buf.m_extentElements.prod() == 0u)
+                            {
+                                throw std::runtime_error("Zero size buffers can not be pinned!");
```

Comment:
Is this necessary to throw an error? We can ignore the pinning call if the memory contains zero elements.
The reason for that is that the memory behind a address is pinned. But there is no memory (zero elements) therefore not pinning is a clean and valid solution.